### PR TITLE
#22 Support ol-ext examples

### PR DESCRIPTION
**Pull request by sanak**
Supports #22.

### CODE REVIEW

**Comment by Siedlerchr** [comment]
Ah that's a good idea!

**Comment by sanak** [comment]
@Siedlerchr (CC: @Tomschi)
Okay, thanks for confirmation!

I will try to check more 3 or 4 examples, then start to write some utility scripts and update `README.md` (or create `EXAMPLES.md`) document in this weekend, so please wait for a while. 😄

File: @types/ol-ext/overlay/Popup.d.ts [hunk 1]
```diff
@@ -120,7 +124,7 @@ export default class Popup extends Overlay {
     popup.show("New informations");
     * @api stable
      */
-    show(coordinate: Coordinate | undefined, features: Feature | Feature[]): void;
+    show(coordinate: Coordinate | string, html: string | undefined): void;
```
Comment (Siedlerchr) [inline @types/ol-ext/overlay/Popup.d.ts:127]:
```suggestion
    show(coordinate: Coordinate | string, html?: string): void;
```
If you use the ? you can omit the undefined (it's just another syntax variant for the same)

File: @types/ol-ext/overlay/Popup.d.ts [hunk 2]
```diff
@@ -39,12 +39,13 @@ export declare type Template = {
...
-    onshow: ((...params: any[]) => any) | undefined;
-    offsetBox: number | number[];
-    positionning: OverlayPositioning | string | undefined;
+    popupClass?: string;
+    anim?: boolean;
+    closeBox?: boolean;
+    onclose?: ((...params: any[]) => any) | undefined;
+    onshow?: ((...params: any[]) => any) | undefined;
+    offsetBox?: number | number[];
+    positioning?: OverlayPositioning | string | undefined;
```
Comment (Siedlerchr) [inline @types/ol-ext/overlay/Popup.d.ts:48]: here you can also remove the undefined if you use the ?

**Comment by Siedlerchr** [comment]
I just merged my other PR with the most recent fixes, hope it doesn't create merge conflicts

File: @types/ol-ext/overlay/Popup.d.ts (hunk 1 above)
Comment (sanak) [reply @types/ol-ext/overlay/Popup.d.ts:127]: @Siedlerchr Okay, thanks for advice!
I will check the original code in `ol-ext`, and some of the code inside the constructor/methods seem to be outdated, so about this part, I will keep the original `undefined` to truck future update possibility.

File: @types/ol-ext/overlay/Popup.d.ts (hunk 2 above)
Comment (sanak) [reply @types/ol-ext/overlay/Popup.d.ts:48]: @Siedlerchr Okay, sure...
I felt that Options class `?` seems to be kind of default (also in `types-ol`), and actually `undefined` is too much, so I will remove `undefined` from here.
Thanks for advice!

**Comment by sanak** [comment]
> I just merged my other PR with the most recent fixes, hope it doesn't create merge conflicts

Okay, thanks for reminder.
I will separate this PR as following PR branches with using git rebase, so it should be no problem.
- `fix/type-defs-from-examples`: Commits only in `@types/ol-ext/**/*.ts.d` (soon).
- `feature/ol-ext-examples (this)`: Based on above branch, and commits except above and @types/jquery trial.

**Comment by sanak** [comment]
@Siedlerchr (CC: @Tomschi)  
I just add the [examples/README.md](https://github.com/gtt-project/types-ol-ext/blob/feature/ol-ext-examples/examples/README.md) on this PR branch, so could you check this ?

Note that `npm run build-examples` will fail without #27 PR modification.

**Comment by Siedlerchr** [comment]
It would be cool if you could push your next fixes directly to this repo (you should have push rights, makes it easier for me to checkout the branch)

**Comment by sanak** [comment]
Okay, sure.
I will push to this repository after rebase from latest master.

**Comment by sanak** [comment]
I recreated this PR (#33) on this repository, so I close this side.

**Pull request closed without merging**